### PR TITLE
Enable CI for cabal-docspec

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -139,6 +145,21 @@ jobs:
...
+      run: |
+        if test -z "$DOCTEST"
+        then
+          mkdir -p $HOME/.cabal/bin
+          curl -sL https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20210111/cabal-docspec-0.0.0.20210111.xz > cabal-docspec.xz
+          echo '0829bd034fba901cbcfe491d98ed8b28fd54f9cb5c91fa8e1ac62dc4413c9562  cabal-docspec.xz' | sha256sum -c -
+          xz -d < cabal-docspec.xz > $HOME/.cabal/bin/cabal-docspec
+          rm -f cabal-docspec.xz
+          chmod a+x $HOME/.cabal/bin/cabal-docspec
+          cabal build all
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:158]: You will have to do `cabal install`.

Comment (rnjtranjan) [reply .github/workflows/haskell.yml:158]: install complain about streamly doesn't have any executables.

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -59,6 +60,7 @@ jobs:
           - 8.6.5+fusion-plugin
           - 8.6.5+streamk
           - 8.4.4+debug
+          - docspec
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:63]: Rename this to `8.10.4+docspec` and move this next to other 8.10.4 tests listed above.

File: .github/workflows/haskell.yml [hunk 3]
```diff
@@ -158,3 +182,5 @@ jobs:
           bin/test.sh --coverage --raw
           hpc-coveralls --repo-token="$COVERALLS_TOKEN" --coverage-mode=StrictlyFullLines
         fi
+
+
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:186]: Remove blank lines.

File: .github/workflows/haskell.yml [hunk 4]
```diff
@@ -139,14 +145,32 @@ jobs:
...
+        then
+          mkdir -p $HOME/.cabal/bin
+          curl -sL https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20210111/cabal-docspec-0.0.0.20210111.xz > cabal-docspec.xz
+          echo '0829bd034fba901cbcfe491d98ed8b28fd54f9cb5c91fa8e1ac62dc4413c9562  cabal-docspec.xz' | sha256sum -c -
+          xz -d < cabal-docspec.xz > $HOME/.cabal/bin/cabal-docspec
+          rm -f cabal-docspec.xz
+          chmod a+x $HOME/.cabal/bin/cabal-docspec
+          cabal build all
+          $HOME/.cabal/bin/cabal-docspec --version
+          $HOME/.cabal/bin/cabal-docspec --timeout 60
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:160]: We can move the cabal build and running docspec in the "Run tests" stanza below.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by harendra-kumar** [comment]
Please squash the commits.

**Comment by rnjtranjan** [comment]
Fixes #1116

File: .github/workflows/haskell.yml [hunk 5]
```diff
@@ -141,12 +147,28 @@ jobs:
...
     - name: Run tests
       run: |
+        if test -n "$DOCTEST"
+        then
+          mkdir -p $HOME/.cabal/bin
+          curl -sL https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20210111/cabal-docspec-0.0.0.20210111.xz > cabal-docspec.xz
+          echo '0829bd034fba901cbcfe491d98ed8b28fd54f9cb5c91fa8e1ac62dc4413c9562  cabal-docspec.xz' | sha256sum -c -
+          xz -d < cabal-docspec.xz > $HOME/.cabal/bin/cabal-docspec
+          rm -f cabal-docspec.xz
+          chmod a+x $HOME/.cabal/bin/cabal-docspec
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:157]: The download part could have remained in the download stanza. But its okay.

File: .github/workflows/haskell.yml [hunk 6]
```diff
@@ -141,12 +147,28 @@ jobs:
...
+          cabal build all
+          $HOME/.cabal/bin/cabal-docspec --version
+          $HOME/.cabal/bin/cabal-docspec --timeout 60
+        fi
+
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:162]: We could use an `else` here, and remove the DOCTEST check below.

**Comment by harendra-kumar** [comment]
I am fixing the docspec failure.

**Comment by harendra-kumar** [comment]
Fixed in #1169 .

**Pull request merged**